pkg/api/defaults-apiserver.go

```diff
@@ -38,9 +38,16 @@ func (cs *ContainerService) setAPIServerConfig() {
 		"--enable-bootstrap-token-auth": "true",
 		"--v":                           "4",
 	}
-	// if using local etcd server then we need the ca file
-	if !(cs.Properties.MasterProfile != nil && cs.Properties.MasterProfile.HasCosmosEtcd()) {
-		staticAPIServerConfig["--etcd-cafile"] = "/etc/kubernetes/certs/ca.crt"
+
+	if cs.Properties.MasterProfile != nil {
+		if cs.Properties.MasterProfile.HasCosmosEtcd() {
+			// Configuration for cosmos etcd
+			staticAPIServerConfig["--etcd-servers"] = fmt.Sprintf("https://%s:%s", cs.Properties.MasterProfile.GetCosmosEndPointURI(), strconv.Itoa(DefaultMasterEtcdClientPort))
+		} else {
+			// Configuration for local etcd
+			staticAPIServerConfig["--etcd-cafile"] = "/etc/kubernetes/certs/ca.crt"
```

Comment:
the logic here is slightly different: before we would set the `--etcd-cafile` when the `MasterProfile` was nil OR if `HasCosmosEtcd`. Now we don't set it for nil `MasterProfile`. The etcd setting is probably not needed for AKS (is it?) but we should be careful about changing AKS logic in a refactor PR.

Also cc @serbrech this is one of those situations where AKS won't get these changes anymore, we need to establish a process for back-porting fixes to the default files.

Reply:
Right. I think this new logic is what we want (UT should tell the tale.)

Definitely worth pointing out to @serbrech although in this case (apiserver config) this doesn't affect AKS at all. But comparable changes to kubelet default config (for example) *will* affect the AKS node kubelet runtime configuration, if AKS doesn't otherwise set all kubelet runtime configuration values explicitly.